rules/TypeDeclaration/Rector/FunctionLike/AddClosureParamTypeFromArgClassStringRector.php

```diff
@@ -75,21 +75,19 @@ public function refactor(Node $node): ?Node
     {
         $this->hasChanged = false;
         foreach ($this->addParamTypeForFunctionLikeParamDeclarations as $addParamTypeForFunctionLikeParamDeclaration) {
-            $type = match (true) {
-                $node instanceof MethodCall => $node->var,
-                $node instanceof StaticCall => $node->class,
-                default => null,
-            };
```

Comment:
In this way we loose type of `$node`, that's why the `if (! ($node->name ?? null) instanceof Identifier) {` was needed.

Instead, the `instanceof` should be used to always improve knowledge about varaible types.